samcli/lib/telemetry/cicd.py

```diff
@@ -0,0 +1,120 @@
+"""
+Module used for detecting whether SAMCLI is executed in a CI/CD environment.
+"""
+
+import os
+from enum import Enum, auto
+from typing import Mapping, Optional, Dict, Union, Callable
+
+
+class CICDPlatform(Enum):
+    Jenkins = auto()
+    GitLab = auto()
+    GitHubAction = auto()
+    TravisCI = auto()
+    CircleCI = auto()
+    AWSCodeBuild = auto()
+    TeamCity = auto()
+    Bamboo = auto()
+    Buddy = auto()
+    CodeShip = auto()
+    Semaphore = auto()
+    Appveyor = auto()
+    # make sure Unknown is at the bottom, it is the fallback.
+    Unknown = auto()
```

Comment:
Maybe rename to `Other`? `Unknown` sounds like we don't know anything, but we do know it's in CI.

Reply:
Good idea, renamed